engine/sound/src/sound.cpp

```diff
@@ -788,7 +788,7 @@ namespace dmSound
 
     bool IsPlaying(HSoundInstance sound_instance)
     {
-        return sound_instance->m_Playing; // && !sound_instance->m_EndOfStream;
+        return sound_instance->m_Playing && !sound_instance->m_EndOfStream;
```

Comment:
Have you run the tests?
The first thing that happened is that it failed.

Also, when running the repro case with this fix, the app crashed.
Did you test your fix?

I don't know _why_ it was commented out, but never removed, but it's slightly more complicated than that. You need to read the code, test it, and reason about why your fix works (and put your reasoning in the PR)